mmv1/products/tpuv2/QueuedResource.yaml

```diff
@@ -107,3 +118,94 @@ properties:
                 - name: 'description'
                   description: |
                     Text description of the TPU.
+                - name: 'networkConfig'
+                  type: NestedObject
+                  description: |
+                    Network configurations for the TPU node.
+                  immutable: true
+                  default_from_api: true
```

Comment:
default_from_api has a couple drawbacks:

- Users won't be able to know accurate information about the end state of the resource (even if it could be known)
- Users can't set the field to its empty/zero value.

I just want to double-check that this field (and the rest marked with default_from_api) don't have [other options](https://googlecloudplatform.github.io/magic-modules/develop/diffs/)?

Reply:
Thanks for highlighting. The [`network` and `subnetwork` fields](https://cloud.google.com/tpu/docs/reference/rest/v2/projects.locations.nodes#Node.NetworkConfig) do get a default value from the API if not set. Consequently, those are subfields of `networkConfig`, which has subfields where some have API defaults and some do not, so I set `networkConfig` to get its default from the API so that its subfields like `network` and `subnetwork` can get these default values. This is also the case for a similar resource TPUv2 VM, [where `networkConfig` has `default_from_api: true`](https://github.com/GoogleCloudPlatform/magic-modules/blob/e831a6f681562d82794af71180e36fc9b568224a/mmv1/products/tpuv2/Vm.yaml#L129) and has been working as expected.

I double checked `queueCount` and that does not get a default value from the TPU API, so I removed `default_from_api` from the `queueCount` subfield here and below.